src/app/dataset/depublication/depublication.component.ts

```diff
@@ -0,0 +1,275 @@
+  /* - close the input dialog
+  */
+  closeDialogs(): void {
+    this.dialogInputOpen = false;
+    this.dialogFileOpen = false;
+  }
+
+  /** setDataFilterParameter
+  /* - update / cancel the data filter parameter
+  /* - trigger polling refresh
+  /*  @param {string} event - the filter information
+   */
+  setDataFilterParameter(event: string): void {
+    console.log('set the search filter ' + event);
+
+    if (event.length === 0) {
+      this.dataFilterParam = undefined;
+    } else {
+      this.dataFilterParam = event;
+    }
+    this.pollingRefresh.next(true);
+  }
+
+  /** setDataSortParameter
+  /* - update / cancel the data sort parameter
+  /* - trigger polling refresh
+  /*  @param {SortParameter} event - the sort information
+   */
+  setDataSortParameter(event: SortParameter): void {
+    if (event.direction === SortDirection.UNSET) {
+      this.dataSortParam = undefined;
+    } else {
+      this.dataSortParam = event;
+    }
+    this.pollingRefresh.next(true);
+  }
+
+  /** toggleOptions
+  /* - toggle the dialog show / hide options
+  */
+  toggleMenuOptions(): void {
+    this.optionsOpen = !this.optionsOpen;
+  }
+
+  /** onSubmitFormFile
+  /* - submit the formFile form data if validation has passed
+  /* - trigger a reload of the displayed depublication data
+  */
+  onSubmitFormFile(): void {
+    const form = this.formFile;
+    if (form.valid) {
+      this.isSaving = true;
+      this.depublications
+        .setPublicationFile(this._datasetId, form.get('depublicationFile')!.value)
+        .subscribe((complete) => {
+          if (complete) {
+            this.pollingRefresh.next(true);
+            this.closeDialogs();
+            this.isSaving = false;
+          }
+        });
+    }
+  }
+
+  /** onSubmitRawText
+  /* - submit the formRawText form data if validation has passed
+  /* - trigger a reload of the displayed depublication data
+  */
+  onSubmitRawText(): void {
+    const form = this.formRawText;
+    if (form.valid) {
+      this.isSaving = true;
+      this.depublications
+        .setPublicationInfo(this._datasetId, form.get('recordIds')!.value.trim())
+        .subscribe(() => {
+          this.pollingRefresh.next(true);
+          form.get('recordIds')!.reset();
+          this.closeDialogs();
+          this.isSaving = false;
+        });
+    }
+  }
+
+  /** beginPolling
+  *  - sets up a timed polling mechanism that only ticks when the last data-result has been retrieved
+  *  - subscribes to the poll
+  /* - instantiates a Subject for poll refreshing
+  */
+  beginPolling(): void {
+    let pushPollDepublication = 0;
+
+    // stream for apply-filter
+    this.pollingRefresh = new Subject();
+    this.pollingRefresh.subscribe(() => {
+      pushPollDepublication += 1;
+    });
+
+    const loadTriggerDepublication = new BehaviorSubject(true);
+
+    const polledDepublicationData = loadTriggerDepublication.pipe(
+      merge(this.pollingRefresh), // user events comes into the stream here
```

Comment:
My IDE says that this merge function is deprecated and should be used in a static way instead.

Reply:
That's probably true.  All the services are singleton classess (does your IDE say it for the others?) that could indeed be static, but the architecture uses the injection of an instance.

Reply:
I think my IDE is referring to the fact that merge is no longer pipeable (if I understand correctly). See https://rxjs-dev.firebaseapp.com/guide/v6/migration.